### PR TITLE
Segue into Minigames: Ports the Mafia Minigame

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/51788
- https://github.com/tgstation/tgstation/pull/52065
- https://github.com/tgstation/tgstation/pull/52066
- https://github.com/tgstation/tgstation/pull/52083
- https://github.com/tgstation/tgstation/pull/52168
- https://github.com/tgstation/tgstation/pull/52243
- https://github.com/tgstation/tgstation/pull/52244
- https://github.com/tgstation/tgstation/pull/52458
- https://github.com/tgstation/tgstation/pull/56199
- https://github.com/tgstation/tgstation/pull/74033
- https://github.com/tgstation/tgstation/pull/74640
- https://github.com/tgstation/tgstation/pull/74819

## About The Pull Request

Ports the Mafia minigame from TG, updated to the (semi)latest version. All PRs contributing have been linked.

## Why It's Good For The Game

Observer minigames have been proven to increase player retention between rounds & deaths, on other servers. 

Considering the current slump in player population during traditionally "high-pop" hours, minigames like these could be helpful at stopping players from leaving the round when they die.

The current extent of my plans only include CTF & Mafia, but I'm hoping making minigames stand out more, as well as a dedicated menu to access minigames instead of the statpanel, will encourage more minigames from other contributors as well.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/1f9a15e6-3d58-4016-8f54-85e2c91d7ae8

</details>

## Changelog
:cl: rkz, tralezab, EOBGames, Rhials, JohnFulpWillard
add: Mafia button to observer menu
add: The Mafia minigame
/:cl:

### CODE REVIEW

**Comment by Nakalon** [comment]
Very interested in seeing this develop

File: code/modules/client/verbs/suicide.dm [hunk 1]
```diff
@@ -256,6 +256,9 @@
 		message_admins("[key_name(src)] (job: [src.job ? "[src.job]" : "None"]) committed suicide at [AREACOORD(src)].")
 
 /mob/living/proc/canSuicide()
+	var/area/A = get_area(src)
+	if(A.area_flags & BLOCK_SUICIDE)
+		to_chat(src, "<span class='warning'>You can't commit suicide here! You can ghost if you'd like.</span>")
```
Comment (itsmeow) [inline code/modules/client/verbs/suicide.dm:261]: this doesn't actually return, meaning it won't do anything

Comment (Tsar-Salat) [reply code/modules/client/verbs/suicide.dm:261]: guh

**Comment by EvilDragonfiend** [comment]
something's wrong with dme. Failed to TM

> Encountered merge conflicts while test merging one or more sources! (Inner exception: Test Merge #10257 at 91427ec conflicted! Conflicting files:
	- /beestation.dme)

File: code/modules/client/verbs/suicide.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/client/verbs/suicide.dm:261]: fixed

**Comment by DatBoiTim** [comment]
Mafia Maps don't seem to be working right. Just had a round where we got sent to the nullspace room(all the rupees and stuff)

**Comment by EvilDragonfiend** [comment]
Sorry man. Can't organise things well these days :skull:

**Comment by EvilDragonfiend** [comment]
welp, I checked the logs, and it's dumped into last year logs. I can't access it now.

**Pull request closed without merging**